opentelemetry/proto/profiles/v1development/profiles.proto

```diff
@@ -144,6 +144,9 @@ message ScopeProfiles {
   // https://opentelemetry.io/docs/specs/otel/schemas/#schema-url
   // This schema_url applies to all profiles in the "profiles" field.
   string schema_url = 3;
+
+  // Lookup table for attributes.
+  repeated opentelemetry.proto.common.v1.KeyValue attribute_table = 4;
```

Comment:
The argument can be made to even move `attribute_table` to `ResourceProfiles` or `ProfilesData` instead of `ScopeProfiles`. If you prefer this, please let me know.

Reply:
Not moving `attribute_table` for the named reason to `ProfilesData` or `ResourceProfiles` sounds good to me. When moving "up" `attributes_table` to `ScopeProfiles` I just looked at it from a protocol perspective and not from a potential synchronization issues of processors.

> Doesn't the same reasoning (reduce amount of duplicate data) also apply to moving the string table?

yes - the same applies to `string_table`. I can add it here as well.

> Also, should we merge https://github.com/open-telemetry/opentelemetry-proto/pull/606 into this PR, otherwise it can be a little confusing (e.g. [doesn't update](https://github.com/florianl/opentelemetry-proto/blob/9a0dee7addf3d815ac996e93806542dce36169c9/opentelemetry/proto/profiles/v1development/profiles.proto#L256) Profile attributes to be references)

To me, these are independent proposals that should be handled and decided on separately. So I didn't merge everything into a big PR. Depending on which PR gets merged first, the other still can be updated.